tests/checks/integration/test_etcd.py

```diff
@@ -114,3 +114,13 @@ def test_bad_config(self):
                                         tags=['url:http://localhost:4001/test/v2/stats/self'])
 
         self.coverage_report()
+
+    def test_closed_port(self):
+        self.assertRaises(Exception,
+                          lambda: self.run_check({"instances": [{"url": "http://localhost:9999"}]}))
```

Comment:
I chose port 9999 as a port that is likely closed.  Not sure if this is a good assumption.  If needed, we could check if the port is closed at the start of the test.